pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -106,33 +107,15 @@ public QueryExecutor openConnectionImpl(HostSpec[] hostSpecs, String user, Strin
       }
     }
 
-    // - the TCP keep alive setting
     boolean requireTCPKeepAlive = PGProperty.TCP_KEEP_ALIVE.getBoolean(info);
 
-    // NOTE: To simplify this code, it is assumed that if we are
-    // using the V3 protocol, then the database is at least 7.4. That
-    // eliminates the need to check database versions and maintain
-    // backward-compatible code here.
-    //
-    // Change by Chris Smith <cdsmith@twu.net>
-
     int connectTimeout = PGProperty.CONNECT_TIMEOUT.getInt(info) * 1000;
 
-    // - the targetServerType setting
-    HostRequirement targetServerType;
-    String targetServerTypeStr = PGProperty.TARGET_SERVER_TYPE.get(info);
-    try {
-      targetServerType = HostRequirement.valueOf(targetServerTypeStr);
-    } catch (IllegalArgumentException ex) {
-      throw new PSQLException(
-          GT.tr("Invalid targetServerType value: {0}", targetServerTypeStr),
-          PSQLState.CONNECTION_UNABLE_TO_CONNECT);
-    }
+    HostRequirement targetServerType = getTargetServerType(info);
 
     SocketFactory socketFactory = SocketFactoryFactory.getSocketFactory(info);
 
-    HostChooser hostChooser =
-        HostChooserFactory.createHostChooser(hostSpecs, targetServerType, info);
+    HostChooser hostChooser = HostChooserFactory.createHostChooser(hostSpecs, targetServerType, info);
```

Comment:
@AlexElin , these changes would conflict with https://github.com/pgjdbc/pgjdbc/pull/844

Would you possibly like to review #844 as well?

Reply:
@vlsi I reverted conflicting changes